PhysiCell/custom_modules/custom.cpp

```diff
@@ -276,8 +417,181 @@ void naive_bcell_phenotype( Cell* pCell, Phenotype& phenotype , double dt ) {
+	bplasma_cell = find_cell_definition( "B_plasma" );
+
+        // antigen variablae
+	std::vector<double> antigenSequence = EMPTY_VECTOR;
+	bplasma_cell->custom_data.add_vector_variable( "antigenSequence", antigenSequence );
+
+        // antibody variable
+	std::vector<double> antibodySequence = EMPTY_VECTOR;
+	bplasma_cell->custom_data.add_vector_variable( "antibodySequence", antibodySequence );
+
+	// update phenotype
+	bplasma_cell->functions.update_phenotype = NULL;
+	//bplasma_cell->functions.update_phenotype = bplasma_cell_phenotype;  // TODO
+}
+
+
+
+// print antibody/antigen sequence
+void printSequence( std::vector<double>& sequence, std::string prefix = "sequence: " ) {
+    for (char element : prefix) printf("%c", element);
+    for (double element : sequence) {
+        if (element == PAD) {
+            printf("{%d}", (int)element);
+        } else {
+            printf("{%c}", (int)element);
+        }
+    }
+    printf("\n");
+}
+
+
+// generate antibody/antigen sequence given actual length of coding sequence, and length of vector
+std::vector<double> generateSequence( int lenSequence ) {
+    int lenPad = LEN_VECTOR_SEQUENCE - lenSequence;
+    std::vector<double> sequence;
+    for (size_t i = 0; i < lenSequence; ++i) {
+        sequence.push_back(ALPHABET[choose_event(probabilities)]);
+    }
+    for (size_t i = 0; i < lenPad; ++i) {
+        sequence.push_back(PAD);
+    }
+    //printSequence(sequence, "Generated sequence: ");
+    return sequence;
+}
+
+
+// Mutate antibody/antigen sequence given sequence and number of mutations (doesn't affect padding)
+// This function can mutate the same letter twice, which can be fixed by changing mutateProbabilities after each mutation
+// May change if necessary
+void mutateSequence( std::vector<double>& sequence, int mutations ) {
+    // Find length of actual sequence without padding
+    size_t sequenceLength = 0;
+    for (const auto& value : sequence) {
+        if (value != PAD) {
+            ++sequenceLength;
+        } else {
+            break;
+        }
+    }
+
+    // Create equal probability vector to choose from the sequence
+    std::vector<double> mutateProbabilities( sequenceLength, 1.0 / sequenceLength );
+
+    // Mutate the sequence
+    for (size_t i = 0; i < mutations; ++i) {
+        sequence[choose_event(mutateProbabilities)] = ALPHABET[choose_event(probabilities)];
+    }
+}
+
+
+// qunatify hamming distance from to antigen antibody sequences of variable size.
+double alignment( Vector_Variable antigenSequence, Vector_Variable antibodySequence , bool verbose = false) {
+
+    // strip input
+    std::vector<double> antigenCode {};
+    std::vector<double> antibodyCode {};
+
+    for (double element : antigenSequence.value) {
+        if (element != PAD) {
+            antigenCode.push_back(element);
+        }
+    }
+    for (double element : antibodySequence.value) {
+        if (element != PAD) {
+            antibodyCode.push_back(element);
```

Comment:
As you promised in the chat, my change has been undone. No matter. I will not comment about code quality in the future. Instead, I shall use PRs as a way to stay up-to-date with the state of the codebase.